src/main/scala/org/broadinstitute/dsde/rawls/jobexec/SubmissionMonitorActor.scala

```diff
@@ -0,0 +1,283 @@
+    workflowRecOption match {
+      case Some(workflowRec) =>
+        WorkflowStatuses.withName(workflowRec.status) match {
+          case WorkflowStatuses.Succeeded =>
+            executionServiceDAO.outputs(workflowRec.externalId, getUserInfo).map(outputs => Option((workflowRec, Option(outputs))))
+
+          case _ => Future.successful(Option((workflowRec, None)))
+        }
+      case None => Future(None)
+
+    }
+  }
+
+  /**
+   * once all the execution service queries have completed this function is called to handle the responses
+   * @param response
+   * @param executionContext
+   * @return
+   */
+  def handleStatusResponses(response: ExecutionServiceStatusResponse)(implicit executionContext: ExecutionContext): Future[StatusCheckComplete] = {
+    response.statusResponse.collect { case Failure(t) => t }.foreach { t =>
+      logger.error(s"Failure monitoring workflow in submission $submissionId", t)
+    }
+
+    datasource.inTransaction { dataAccess =>
+      val updatedRecs = response.statusResponse.collect {
+        case Success(Some((updatedRec, _))) => updatedRec
+      }
+
+      val workflowsWithOutputs = response.statusResponse.collect {
+        case Success(Some((workflowRec, Some(outputs)))) => (workflowRec, outputs)
+      }
+
+      // to minimize database updates do 1 update per status
+      DBIO.seq(updatedRecs.groupBy(_.status).map { case (status, recs) =>
+        dataAccess.workflowQuery.findWorkflowByIds(recs.map(_.id)).map(r => (r.status, r.statusLastChangedDate)).update(status, new Timestamp(System.currentTimeMillis()))
+      }.toSeq: _*) andThen
+        handleOutputs(workflowsWithOutputs, dataAccess) andThen
+        checkOverallStatus(dataAccess)
+    } map { shouldStop => StatusCheckComplete(shouldStop) }
+  }
+
+  /**
+   * When there are no workflows with a running status, mark the submission as done or aborted as appropriate.
+   * @param dataAccess
+   * @param executionContext
+   * @return true if the submission is done/aborted
+   */
+  def checkOverallStatus(dataAccess: DataAccess)(implicit executionContext: ExecutionContext): ReadWriteAction[Boolean] = {
+    dataAccess.workflowQuery.listWorkflowRecsForSubmissionAndStatuses(submissionId, WorkflowStatuses.runningStatuses:_*) flatMap { workflowRecs =>
+      if (workflowRecs.isEmpty) {
+        dataAccess.submissionQuery.findById(submissionId).map(_.status).result.head.map { status =>
+          SubmissionStatuses.withName(status) match {
+            case SubmissionStatuses.Aborting => SubmissionStatuses.Aborted
+            case SubmissionStatuses.Submitted => SubmissionStatuses.Done
+            case _ => throw new RawlsException(s"submission $submissionId in unexpected state $status, expected Aborting or Submitted")
+          }
+        } flatMap { newStatus => dataAccess.submissionQuery.updateStatus(submissionId, newStatus) } map(_ => true)
```

Comment:
there is a debug log in the receive of the actor... do you think that is good enough?